sqlmesh/core/engine_adapter/base.py

```diff
@@ -261,6 +262,16 @@ def set_current_catalog(self, catalog: str) -> None:
         """Sets the catalog name of the current connection."""
         raise NotImplementedError()
 
+    def get_catalog_type(self, catalog: t.Optional[str]) -> str:
+        """Intended to be overridden for data virtualization systems like Trino that,
+        depending on the target catalog, require slightly different properties to be set when creating / updating tables
+        """
+        return self.DEFAULT_CATALOG_TYPE
+
+    @property
+    def current_catalog_type(self) -> str:
+        return self.get_catalog_type(self.get_current_catalog())
```

Comment:
It seems like we should raise `UnsupportedCatalogOperationError` here if `self.CATALOG_SUPPORt.is_unsupported`. It is strange for example for MySQL to return a `mysql` current catalog when it doesn't support catalogs at all.

Reply:
Fair call, it didn't even occur to me that catalogs might not be supported by some databases